Entities/Common/Scripts/AlignToGround.as

```diff
@@ -0,0 +1,25 @@
+
+void onInit(CBlob@ this)
+{
+	CShape@ shape = this.getShape();
+	
+	// put on ground based on shape's position
+	CMap@ map = getMap();
+	uint ts = map.tilesize;
+	Vec2f pos = shape.getPosition();
+	pos = Vec2f(Maths::Floor(pos.x / ts) * ts, Maths::Floor(pos.y / ts) * ts); // align to tiles
+
+	while (!map.isTileSolid(pos))
+	{
```

Comment:
My only suggestion is to add a failsafe to this loop, maybe stopping after trying to shift the tent up 10 times. Mappers may accidentally cause this to loop forever when making maps